w3-blob.md

```diff
@@ -0,0 +1,419 @@
+Accept --o Accepted
+Accepted -- claim --> Claim
+Added -- claim --> Accept
+```
+
+**Iconography**
+
+- Icon on the left describes type of the node
+- Icon on the right describes issuer / audience pair if only one audience is the issuer
+- ▶️ Task
+- ⏭️ Next Task (a.k.a Effect)
+- 🧾 Receipt
+- 🎫 Delegation / Commitment
+- 🚦 Await
+- 👩‍💻 Alice
+- 🤖 Service
+
+### Add Blob Invocation Example
+
+Shown Invocation example illustrates Alice requesting to add 2MiB blob to her space.
+
+```js
+{
+  "cmd": "/space/content/add/blob",
+  "sub": "did:key:zAlice",
+  "iss": "did:key:zAlice",
+  "aud": "did:web:web3.storage",
+  "args": {
+    "blob": {
+      // multihash of the blob as byte array
+      "content": { "/": { "bytes": "mEi...sfKg" } },
+      // size of the blob in bytes
+      "size": 2_097_152,
+    }
+  }
+}
+```
+
+### Add Blob Receipt Example
+
+Shows an example receipt for the above `/space/content/add/blob` capability invocation.
+
+> ℹ️ We use `// "/": "bafy..` comments to denote CID of the parent object.
+
+```js
+{ // "/": "bafy..add",
+  "iss": "did:web:web3.storage",
+  "aud": "did:key:zAlice",
+  "cmd": "/ucan/assert/result"
+  "sub": "did:web:web3.storage",
+  "args": {
+    // refers to the invocation from the example
+    "ran": { "/": "bafy..add" },
+    "out": {
+      "ok": {
+        // result of the add is the content (location) claim
+        // that is produced as result of "bafy..accept"
+        "claim": { "await/ok": { "/": "bafy...accept" } }
+      }
+    },
+    // Previously `next` was known as `fx` instead, which is
+    // set of tasks to be scheduled.
+    "next": [
+      // 1. System attempts to allocate memory in user space for the blob.
+      { // "/": "bafy...alloc",
+        "cmd": "/service/blob/allocate",
+        "sub": "did:web:web3.storage",
+        "args": {
+          // space where memory is allocated
+          "space": "did:key:zAlice",
+          "blob": {
+            // multihash of the blob as byte array
+            "content": { "/": { "bytes": "mEi...sfKg" } },
+            // size of the blob in bytes
+            "size": 2_097_152,
+          }
+        }
+      },
+      // 2. System will attempt to accept received content
+      // if matches blob multihash and size.
+      { // "/": "bafy...accept",
+        "cmd": "/service/blob/accept",
+        "sub": "did:web:web3.storage",
+        "args": {
+          "space": "did:key:zAlice",
+          "blob": {
+            // multihash of the blob as byte array
+            "content": { "/": { "bytes": "mEi...sfKg" } },
+            "size": 2_097_152,
+          },
+          // This task is blocked on allocation
+          _: { "await/ok": { "/": "bafy...alloc" } }
+        }
+      }
+    ]
+  }
+}
+```
+
+### Add Blob Capability
+
+#### Add Blob Capability Schema
+
+```ts
+type AddBlob = {
+  cmd: "/space/content/add/blob"
+  sub: SpaceDID
+  args: {
+    blob: Blob
+  }
+}
+
+type Blob = {
+  content: Multihash
+  size:    int
+}
```

Comment:
We should allow user specify an expiry here so we can reflect it on our end with `accept`.

Reply:
I think you may have misunderstood my intention behind `exp` in `accept`, there it meant to capture pre-signed url expiry. While here I would expect it to imply that blob should be GC-ed at given time.

Am I interpreting your intentions correctly here ? If so I'd say lets not add this until we actually are able to support GC-ing blobs, when we do we can introduce this field.

Reply:
yes, I was considering this to be the expiration we talked about for expiration of claim https://github.com/web3-storage/specs/pull/115#discussion_r1530787111

but than can come as follow up